File: 0415_easy_hot_add-strings/solution.java
class Solution {
    public String addStrings(String num1, String num2) {
        StringBuilder sb = new StringBuilder();

        int i = num1.length() - 1;
        int j = num2.length() - 1;
        int carry = 0;

        while (i >= 0 || j >= 0 || carry > 0) {
            int a = i >= 0 ? num1.charAt(i) - '0' : 0;
            int b = j >= 0 ? num2.charAt(j) - '0' : 0;

            sb.append(String.valueOf((a+b+carry) % 10));
            carry = (a+b+carry) / 10;

            i--;
            j--;

        }

        return sb.reverse().toString();
    }
}

File: 0415_easy_hot_add-strings/solution.py
class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        result = ''

        carry = 0
        i = len(num1) - 1
        j = len(num2) - 1

        while i >= 0 or j >= 0 or carry:
            a = int(num1[i]) if i >=0 else 0
            b = int(num2[j]) if j >=0 else 0
            result = str((a+b+carry) % 10) + result
            carry = (a+b+carry) // 10
            i -= 1
            j -= 1
        
        return result
        